### ISSUE TITLE
Benchmark Netdata dbengine with Time Series Benchmark Suite

### ISSUE THREAD

**Issue by mfundul** [netdata/netdata#6213]
We should benchmark the Netdata dbengine with [Time Series Benchmark Suite](https://blog.timescale.com/time-series-database-benchmarks-timescaledb-influxdb-cassandra-mongodb-bc702b72927e/) to compare its performance with other DB engines and also catch any performance regressions. The github page of TSBS is [here](https://github.com/timescale/tsbs).

We should do research on how to bypass the traditional Netdata codepath and access the dbengine API directly from the benchmark.

**Comment by ilyam8** [comment]
> To accomplish this in a fair way, the data to be inserted and the queries to run are pre-generated and native Go clients are used wherever possible to connect to each database (e.g., mgo for MongoDB).

`influxdb` has [http api](https://docs.influxdata.com/influxdb/v1.7/tools/api/) which u is used by `Time Series Benchmark Suite`. It would be nice if netdata has something similar.

But is it fair to compare `dbengine` with other TSDBs? is dbengine a TSDB? (stores time series; a time series is a series of data points indexed in time order).

So those tests not only for load/query DB, but load DB with time series data and query the data.

**Comment by cakrit** [comment]
Yes, we'll need to add such an API (useful for many things actually, also unit and load tests). 
And yes, it is essentially a TSDB, with perhaps a couple of API calls missing (@mfundul had already mentioned something like "give me the 10 next entries").

**Comment by cakrit** [comment]
For the "query" part, can you use the `/api/v1/data` call? See https://editor.swagger.io/?url=https://raw.githubusercontent.com/netdata/netdata/master/web/api/netdata-swagger.yaml

If you can, then we just need a new `/api/v1/write` call. 

@mfundul you'll need to add the new call to the array in https://github.com/netdata/netdata/blob/master/web/api/web_api_v1.c#L767
It will be `{ "manage/dbengine",   0, WEB_CLIENT_ACL_MGMT,      web_client_api_request_v1_mgmt_dbengine     }`

The implementation of the function `web_client_api_request_v1_mgmt_dbengine` should be under `web/api/dbengine`. See `web/api/health/health_cmdapi.c` for an example. 
Especially important there is the additional security via the auth bearer token. See the code under 
`https://github.com/netdata/netdata/blob/master/web/api/health/health_cmdapi.c#L67`

@ilyam8 the precise method of using the auth bearer token in the calls is described [here](https://docs.netdata.cloud/web/api/health/#health-management-api). You can also check [the health mgmt api test script](https://github.com/netdata/netdata/blob/master/tests/health_mgmtapi/health-cmdapi-test.sh.in).

**Comment by mfundul** [comment]
This is blocked until we implement variable granularity metrics, data blobs and metrics map.
